lightning/src/offers/invoice_request.rs

```diff
@@ -998,15 +998,37 @@ impl VerifiedInvoiceRequest {
 		InvoiceRequestFields {
 			payer_signing_pubkey: *payer_signing_pubkey,
 			quantity: *quantity,
-			payer_note_truncated: payer_note.clone().map(|mut s| {
-				s.truncate(PAYER_NOTE_LIMIT);
-				UntrustedString(s)
-			}),
+			payer_note_truncated: payer_note
+				.clone()
+				// Truncate the payer note to `PAYER_NOTE_LIMIT` bytes, rounding
+				// down to the nearest valid UTF-8 code point boundary.
+				.map(|s| UntrustedString(string_truncate_safe(s, PAYER_NOTE_LIMIT))),
```

Comment:
What is the purpose of this truncation, is it a requirement? I was searching through the bolt spec, but couldn't find this limit.

If the limit is needed but internal, maybe the string can be truncated to a fixed number of unicode characters (`chars().take(max_chars)` ?), if that's simpler.

Reply:
Got it. This is because LDK is stateless and there is no invoice database right?